connectivity/service/src/main/java/org/eclipse/ditto/connectivity/service/messaging/OutboundMappingProcessor.java

```diff
@@ -246,35 +253,109 @@ private List<MappingOutcome<OutboundSignal.Mapped>> processMappableSignals(
             signalToMap = outboundSignalSource;
         }
 
-        final Adaptable adaptableWithoutExtra = mappingTimer.protocol(() -> protocolAdapter.toAdaptable(signalToMap));
-        final Adaptable adaptable = outboundSignal.getExtra()
-                .map(extra -> ProtocolFactory.setExtra(adaptableWithoutExtra, extra))
-                .orElse(adaptableWithoutExtra);
-
-        final var adaptableWithInternalCorrelationId = mappableSignals.stream()
-                .findFirst()
-                .map(signal -> setInternalCorrelationIdToAdaptable(adaptable, signal.getSource()))
-                .orElse(adaptable);
+        final boolean hasPartialAccessPaths = dittoHeaders.containsKey(DittoHeaderDefinition.PARTIAL_ACCESS_PATHS.getKey());
 
         return mappingTimer.overall(() -> mappableSignals.stream()
                 .flatMap(mappableSignal -> {
-                    final Signal<?> source = mappableSignal.getSource();
+                    final Signal<?> source = signalToMap;
```

Comment:
@hu-ahmed I don't think this change is right .. 
This should be `mappableSignal.getSource();` - because in a payload mapping, a single "signal" can cause multiple signals to be published.

E.g. a JavaScript payload mapping can create 10 messages to be published for a single event.  
All of those would need to be checked/filtered for permissions.